core/unit_test/TestTeam.hpp

```diff
@@ -1338,11 +1338,12 @@ struct TestTeamBroadcast<
   template <class ScalarType>
   static inline
       typename std::enable_if<!std::is_integral<ScalarType>::value, void>::type
-      compare_test(ScalarType A, ScalarType B) {
-    if (std::is_same<ScalarType, double>::value) {
-      ASSERT_DOUBLE_EQ((double)A, (double)B);
-    } else if (std::is_same<ScalarType, float>::value) {
-      ASSERT_FLOAT_EQ((double)A, (double)B);
+      compare_test(ScalarType A, ScalarType B, double epsilon_factor) {
+    if (std::is_same<ScalarType, double>::value ||
+        std::is_same<ScalarType, float>::value) {
+      ASSERT_NEAR((double)A, (double)B,
+                  epsilon_factor * std::abs(A) *
+                      std::numeric_limits<ScalarType>::epsilon());
```

Comment:
I am not a big fan of passing `epsilon_factor`, I'd pass directly the relative tolerance.

In any case, comment on what you are doing.

Reply:
Isn't the comment below enough? The reason for the split is that I want to keep the float vs not float specifity etc. here in the comparison, while the test has to pass in the algorithmic epsilon. Hence my comment about the random walk.

Reply:
I can't really tell from the comment below nor the name of the function that this how you will use the trailing argument.  That said the comparison function is defined right above `TestTeamBroadcast` so it is not a big deal if you don't want to change.

Reply:
I think the deciding factor here is whether we think we could ever want a different epsilon_factor for different types, which I don't think we ever will.